Add core parts of mistweaver guide

trevorm4 opened the pull request:
Adds core abilities, short cooldowns, and healing cd sections. The healing cd/short cooldown section are not complete quite yet, but want to split up the PRs to make it easier to review

example report: report/7VQTDLB24m6NHMAF/9-Mythic+Raszageth+the+Storm-Eater+-+Kill+(13:26)/Sweggles/standard/overview

![image](https://user-images.githubusercontent.com/11250934/218353061-f9f8c84f-a9d2-442d-b917-fcf42762f142.png)
![image](https://user-images.githubusercontent.com/11250934/218353089-0cfff041-27dc-43df-aace-519c6c3c9517.png)

emallson commented on src/analysis/retail/monk/mistweaver/modules/spells/ThunderFocusTea.tsx:
```diff
@@ -168,6 +193,58 @@ class ThunderFocusTea extends Analyzer {
+          <div>
+            {this.subStatistic()} <br />
+            <small>
+              Green indicates a correct <SpellLink id={TALENTS_MONK.THUNDER_FOCUS_TEA_TALENT} />{' '}
+              cast, while red indicates a bad cast.
```
this should either be "correct/incorrect" or "good/bad", not "correct/bad" or "good/incorrect".

i would generally avoid "correct/incorrect" personally (also applies to the tooltips above) but i'm not a stickler on that. personal preference

emallson reviewed (commented):
minor wording comment. code seems fine

emallson commented:
not a huge fan of how the cast efficiency charts look with very low cooldown spells like RSK. something to work on later probably 🤔

vohrr commented on src/analysis/retail/monk/mistweaver/modules/spells/EssenceFontTargetsHit.tsx:
```diff
@@ -35,6 +37,17 @@ class EssenceFontTargetsHit extends Analyzer {
+  getPerformance(targetsHit: number) {
+    if (targetsHit < 12) {
+      return QualitativePerformance.Fail;
+    } else if (targetsHit < 14) {
```
Average targets hit per cast is rarely above is 14, I would bump these down.
This also doesn't account for Upwelling bolts, you could do something like make these values a percentage of expected bolts - which is already calc'ed in cancelled casts

vohrr commented on src/analysis/retail/monk/mistweaver/modules/spells/RenewingMist.tsx:
```diff
@@ -51,6 +63,49 @@ class RenewingMist extends Analyzer {
+        <SpellLink id={TALENTS_MONK.RENEWING_MIST_TALENT} /> as much as possible is extremely
+        important due to its synergy with <SpellLink id={SPELLS.VIVIFY} />,{' '}
+        <SpellLink id={TALENTS_MONK.RISING_MIST_TALENT} />, and{' '}
+        <SpellLink id={TALENTS_MONK.MISTY_PEAKS_TALENT} />
```
Dancing mists

vohrr commented on src/analysis/retail/monk/mistweaver/modules/spells/RisingSunKick.tsx:
```diff
@@ -46,6 +52,45 @@ class RisingSunKick extends Analyzer {
+        <b>
+          <SpellLink id={TALENTS_MONK.RISING_SUN_KICK_TALENT.id} />
+        </b>{' '}
+        is one of your primary damaging spells and also an important healing spell due to{' '}
```
'is one of your primary damaging spells but is also the highest priority healing spell along side renewing mist' 

-- want to stress the importance of this and ReM as much possible simply because 90% of logs fail these two metrics more than anything else

trevorm4 commented:
> not a huge fan of how the cast efficiency charts look with very low cooldown spells like RSK. something to work on later probably 🤔

I agree to be honest, I think we might want to have an optional parameter to omit the cast lines and only show the shadings maybe?

trevorm4 commented:
![image](https://user-images.githubusercontent.com/11250934/218622961-468a3999-40f9-4643-bda4-1da97a80805e.png)
![image](https://user-images.githubusercontent.com/11250934/218623424-5cd6ca17-6f85-411f-8b5a-2496de5d5672.png)

Updated appearance, changed some text, and changed EF thresholds.
